presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -210,7 +211,13 @@ public BytecodeNode generateExpression(Signature signature, BytecodeGeneratorCon
                         .comment("lookupSwitch(hashCode(<stackValue>))")
                         .dup(javaType)
                         .append(invoke(hashCodeBinding, hashCodeSignature))
+                        //.invokeStatic(Types.class, "longHashToShortHash", int.class, long.class)
```

Comment:
I was wondering which version is preferred here as the pattern in code is not clear (some parts are generated to bytecode and some are invoked). If you prefer other solution I don't see any obvious downside of that.